#12550 - Hide jurisdiction fields when configured

MateStrysewske opened the pull request:
<!--
If you've never submitted a pull request to the SORMAS repository before or this is your first time using this template, please read the Contributing guidelines (https://github.com/sormas-foundation/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) for an explanation of our guidelines regarding pull requests. You don't have to remove this comment or from your pull request as it will automatically be hidden.

Please specify the number of the issue this pull request is related to after the #.
-->
Fixes #12550

sormas-vitagroup commented:
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12618

sergiupacurariu commented:
the server cannot be started because in the StartupShutdownService the "createConstantFacilities()" method runs before "upgrade()" method this leads to an error as the "defaultInfrastructure" field does not exist yet.

sergiupacurariu commented:
- In the Configuration menu is not possible to check whether there is or not a default value set for an infrastructure type. It is very unpleasant to change it since the only way to do it is to 
  find it and deselect it then go and select default a new one.
  - Environment forms created and edited have to be modified the same way as for cases. Currently, the facilities available in the form are connected to the user's jurisdiction and not related to the default infrastructure set in Configuration.
  - The same changes are needed on the Environment Samples form as on the Environment forms.
  - it is possible to set default infrastructure that is not related, meaning you can choose default Region1 and default District from different region. On case save all the Infrastructure fields on the case (region_id, district_id, community_id) will be null. After saving, the case cannot be seen in the case grid. Even when the default infrastructure is selected consistently (e.g. default District is from the default Region) the case is saved with empty infrastructure fields.
  - Person grid - no persons are seen in the grid despite the creation of a few cases that also have a person who should be accessible

MateStrysewske commented:
@sergiupacurariu 

- Only regions, districts and communities can be default infrastructures now
- Added a column to those grids so users can see which is the default
- Added validation that prevents e.g. a district being set as default when it's not in the default region

Apart from that:
- Android doesn't need to be covered
- Environments don't need to be covered for now; I will add a respective warning to the Wiki and release notes
- It's also fine for now that users won't see data entered by them if they're not in the default jurisdiction, because this is not how this feature is supposed to be used

This is still pretty messy and I'd like to have it more consistent, but that would mean A LOT more effort that won't really be necessary at this point.

sergiupacurariu commented on sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java:
```diff
@@ -3141,6 +3141,12 @@ public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource, int
 				getDao(PathogenTest.class)
 					.executeRaw("ALTER TABLE pathogentest ADD COLUMN testedPathogen varchar(255);");
 
+			case 353:
```
this version needs to be updated to 354

sergiupacurariu commented:
- on all three Region, District and Community grids - the defaultInfrastructure field is not added in the sorting properties so filtering by Default crashes the grid
  - the defaultInftrastructureCache is not populated after the default Region is saved and also when the default region is changed conducting to an error while trying to save a district from the default Region. Because of this I even managed to save a district that was from the old region which was still stored in the cache leading to a situation of having a default district from outside the default region.
  - if there is a default district and community set it is possible to remove the default district while we still have a default community (without a default district). After this, it is also possible to save a new default district that is different from the default community district.

sormas-vitagroup commented:
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12618

MateStrysewske commented:
@sergiupacurariu Updated DatabaseHelper. Yes, that needs to be mentioned, but since these changes are only on dev until the next release, it doesn't make sense to add these hints right now. I'll add them to the release notes and the Wiki page during the release.

sormas-vitagroup commented:
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12618
